### PR TITLE
Node 1249: EVALUATED instead of Any

### PR DESCRIPTION
**Pull request by ismagin**

### CODE REVIEW

**Comment by ismagin** [comment]
benchmark tests:
```
BEFORE:

[info] Benchmark                              Mode  Cnt         Score         Error  Units
[info] ScriptEvaluatorBenchmark.base58decode  avgt   10   1463757.373 ▒   89902.107  ns/op
[info] ScriptEvaluatorBenchmark.base58encode  avgt   10   4185553.212 ▒  142225.969  ns/op
[info] ScriptEvaluatorBenchmark.bigSum        avgt   10    449198.151 ▒    4250.903  ns/op = 450 microsec/op
[info] ScriptEvaluatorBenchmark.bytesConcat   avgt   10    947695.268 ▒   48272.662  ns/op
[info] ScriptEvaluatorBenchmark.nestedBlocks  avgt   10   1455389.961 ▒   18584.915  ns/op
[info] ScriptEvaluatorBenchmark.signatures    avgt   10  11952575.867 ▒  140917.423  ns/op
[info] ScriptEvaluatorBenchmark.stringConcat  avgt   10  30002940.746 ▒ 4172020.587  ns/op


With EVALUATED:

[info] Benchmark                              Mode  Cnt         Score         Error  Units
[info] ScriptEvaluatorBenchmark.base58decode  avgt   10   1654749.371 ▒  133805.423  ns/op
[info] ScriptEvaluatorBenchmark.base58encode  avgt   10   4721028.190 ▒ 1018519.919  ns/op
[info] ScriptEvaluatorBenchmark.bigSum        avgt   10    456104.767 ▒   29266.197  ns/op = 450 microsec/op
[info] ScriptEvaluatorBenchmark.bytesConcat   avgt   10    875302.999 ▒   15638.125  ns/op
[info] ScriptEvaluatorBenchmark.nestedBlocks  avgt   10   1487708.692 ▒    9504.776  ns/op
[info] ScriptEvaluatorBenchmark.signatures    avgt   10  11894626.516 ▒   38365.854  ns/op
[info] ScriptEvaluatorBenchmark.stringConcat  avgt   10   8937312.704 ▒   81495.399  ns/op

[info] Benchmark                              Mode  Cnt         Score        Error  Units
[info] ScriptEvaluatorBenchmark.base58decode  avgt   10   1523602.002 ▒  18082.178  ns/op
[info] ScriptEvaluatorBenchmark.base58encode  avgt   10   4539273.067 ▒ 562990.931  ns/op
[info] ScriptEvaluatorBenchmark.bigSum        avgt   10    440274.278 ▒   2630.782  ns/op
[info] ScriptEvaluatorBenchmark.bytesConcat   avgt   10    877292.129 ▒  10851.531  ns/op
[info] ScriptEvaluatorBenchmark.nestedBlocks  avgt   10   1491328.958 ▒  12890.307  ns/op
[info] ScriptEvaluatorBenchmark.signatures    avgt   10  11884075.279 ▒  91497.729  ns/op
[info] ScriptEvaluatorBenchmark.stringConcat  avgt   10   8906253.111 ▒  61414.042  ns/op
```

**Comment by ismagin** [comment]
TLDR: All the same, string concat became MUCH faster, probably result of removing
```
  lazy val sumString: BaseFunction = createRawOp(SUM_OP, STRING, STRING, SUM_STRING, "Limited strings concatination", "prefix", "suffix", 10)((a, b) => {
    lazy val astr = a.asInstanceOf[String]
    lazy val bstr = b.asInstanceOf[String]
    lazy val al   = astr.length
    lazy val bl   = bstr.length
    Either.cond(al + bl <= MaxStringResult, astr + bstr, "String is too large")
  })
```
and replacing with plain 
```
lazy val sumString: BaseFunction =
    createRawOp(SUM_OP, STRING, STRING, SUM_STRING, "Limited strings concatination", "prefix", "suffix", 10) {
      case (CONST_STRING(a), CONST_STRING(b)) =>
        lazy val al = a.length
        lazy val bl = b.length
        Either.cond(al + bl <= MaxStringResult, CONST_STRING(a + b), "String is too large")
      case _ => ???
    }
```

mind removing `asInstanceOf[String]`
